gcp_variant_transforms/transforms/extract_input_size.py

```diff
@@ -0,0 +1,134 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A PTransform to output a PCollection of ``Variant`` records to BigQuery."""
+
+from __future__ import absolute_import
+
+import apache_beam as beam
+
+from apache_beam.io import filesystems
+from apache_beam.typehints import Any, List, Iterable
+from apache_beam.typehints import with_input_types
+from apache_beam.typehints import with_output_types
+from gcp_variant_transforms.beam_io import vcf_estimate_io
+
+
+@beam.typehints.with_input_types(vcf_estimate_io.VcfEstimate)
+class ExtractLineCount(beam.DoFn):
+  def process(self, estimate):
+    yield estimate.estimated_line_count
+
+
+@beam.typehints.with_input_types(vcf_estimate_io.VcfEstimate)
+class ExtractFileSize(beam.DoFn):
+  def process(self, estimate):
+    yield estimate.size_in_bytes
+
+
+@with_input_types(List[int])
+@with_output_types(int)
+class SumEstimations(beam.CombineFn):
+  """CombineFn for computing PCollection size."""
+
+  def create_accumulator(self):
+    return 0
+
+  def add_input(self, accumulator, element):
+    return accumulator + element
+
+  def add_inputs(self, accumulator, elements):
+    return sum(elements) + accumulator
+
+  def merge_accumulators(self, accumulators):
+    return sum(accumulators)
+
+  def extract_output(self, accumulator):
+    return accumulator
+
+
+@with_input_types(Iterable[Any])
+@with_output_types(int)
+class SumRecordCounts(beam.CombineFn):
+  """CombineFn for computing PCollection size."""
+
+  def create_accumulator(self):
+    return 0
+
+  def add_input(self, accumulator, element):
+    return accumulator + sum(element)
+
+  def add_inputs(self, accumulator, elements):
+    return accumulator + sum(map(sum, elements))
+
+  def merge_accumulators(self, accumulators):
+    return sum(accumulators)
+
+  def extract_output(self, accumulator):
+    return accumulator
+
+
+class GetFilesSize(beam.PTransform):
+  def expand(self, estimates):
+    return (estimates
+            | 'ExtractFileSize' >> beam.ParDo(ExtractFileSize())
+            | 'SumFileSizes' >> beam.CombineGlobally(SumEstimations()))
+
+
+class GetEstimatedLineCount(beam.PTransform):
+  def expand(self, estimates):
+    return (estimates
+            | 'ExtractLineCount' >> beam.ParDo(ExtractLineCount())
+            | 'SumLineCounts' >> beam.CombineGlobally(SumEstimations()))
+
+
+class GetSampleMap(beam.PTransform):
+  def _get_call_names(self, estimate):
+    # type: (vcf_parser.Variant) -> Tuple[str]
+    """Returns the names of all calls for the variant."""
+    return tuple(
+        zip(estimate.samples,
+            [estimate.estimated_line_count] * len(estimate.samples)))
+
+  def expand(self, estimates):
+    return (estimates
+            | 'MapSamplesToRecordCount' >> beam.FlatMap(self._get_call_names)
+            | 'GroupAllSamples' >> beam.GroupByKey())
+
+
+class GetEstimatedRecordCount(beam.PTransform):
```

Comment:
I found the naming is confusion.

Reply:
As discussed offline, the number of records can be simplified as number of samples * number of variants per file. But I will leave the decision to you. And also, some comments/documentation might need to help understand the `record` here.

Reply:
Added documentation to the module and to the SampleMap.

I think this is the most efficient approach, to remove any unnecessary double counting.

Reply:
I would usually refer to this as a "cell" count (rows * columns = cells)

Unfortunately, here "cells" might be overloaded.

Remind me why this value is useful?  When do we actually care how many total cells there are?  [I know we talked about it I just can't recall]

Reply:
Can we just call it Value count?  eg GetEstimatedValueCount?

Reply:
Done.